src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -89,7 +90,7 @@ public class DistributedSession : ISession
             _idleTimeout = idleTimeout;
             _ioTimeout = ioTimeout;
             _tryEstablishSession = tryEstablishSession;
-            _store = new Dictionary<EncodedKey, byte[]>();
+            _store = new DefaultDistributedSessionStore();
```

Comment:
???

Reply:
We use a subset of the dictionary APIs in this type. Some of the unused APIs result in weird choices with nullability. I weighed nullable forgiveness, throwing NotSupported and just creating a more focused abstraction and the last seemed the best.

Reply:
This deserved a mega comment as its kinda snuck in with this change and there's nothing describing the why

Reply:
Yup I should have called this out. @Tratcher and I have been discussing this in a different comment thread, so I'd hoped he would have magically understood this

Reply:
?

Reply:
Let's start from scratch. `DistributedSession` uses a dictionary store although not all of the dictionary APIs are used. When the initial Load throws an exception, it uses a fallback store that no-ops. With nullable enabled, the current implementation of the no-op store falls foul. It returns `null` values for implementations that should not.

There are a couple of options we could use here:

a) Use null-forgiveness 
b) Have the API return a non-null value (`Array.Empty<byte>()`)
c) Realize that `DistributedSession` never invokes these APIs and create a smaller abstraction to work with that's null-correct

Option (c) worked best which is what the change does. The current implementation uses `IDictionary<,>` when referencing the backing field, so we're not introducing new runtime costs via callvirts or boxing the dictionary enumerator.

Reply:
I see.... Please add this comment to DefaultDistributedSessionStore so we don't just delete it.

Reply:
```suggestion
            // When using a NoOpSessionStore, using a dictionary as a backing store results in problematic API choices particularly with nullability.
            // We instead use a more limited contract - `IDistributedSessionStore` as the backing store that plays better.
            _store = new DefaultDistributedSessionStore();
```